pkg/tm/transaction_executor.go

```diff
@@ -80,23 +80,40 @@ func Begin(ctx context.Context, name string) context.Context {
 
 // commit global transaction
 func CommitOrRollback(ctx context.Context, err *error) error {
+	role := *GetTransactionRole(ctx)
+	if role == PARTICIPANT {
+		// Participant has no responsibility of rollback
+		log.Debugf("Ignore Rollback(): just involved in global transaction [%s]", GetXID(ctx))
+		return nil
+	}
 	tx := &GlobalTransaction{
 		Xid:    GetXID(ctx),
 		Status: *GetTxStatus(ctx),
-		Role:   *GetTransactionRole(ctx),
+		Role:   role,
 	}
+	var (
+		resp error
+		// retry times should read from config
+		retry = 10
+	)
 
-	var resp error
-	if *err == nil {
-		resp = GetGlobalTransactionManager().Commit(ctx, tx)
-		if resp != nil {
-			log.Infof("transactionTemplate: commit transaction failed, error %v", err)
+	for {
+		if err == nil || *err == nil {
+			resp = GetGlobalTransactionManager().Commit(ctx, tx)
+			if resp != nil {
+				log.Infof("transactionTemplate: commit transaction failed, error %v", err)
+			}
+		} else {
+			resp = GetGlobalTransactionManager().Rollback(ctx, tx)
+			if resp != nil {
+				log.Infof("transactionTemplate: Rollback transaction failed, error %v", err)
+			}
 		}
-	} else {
-		resp = GetGlobalTransactionManager().Rollback(ctx, tx)
-		if resp != nil {
-			log.Infof("transactionTemplate: Rollback transaction failed, error %v", err)
+		retry--
+		if resp == nil || retry == 0 {
+			break
 		}
 	}
+	// todo unbind xid
 	return resp
 }
```

Comment:
```go
txManager := GetGlobalTransactionManager()
if txManager == nil {
     return perr("xxxxxx")
}

if commitErr := txManager.Commit(ctx, tx); comitter == nil {
    return nil
}

retry := 10
for retry > 0 {
    if rollbackErr := GetGlobalTransactionManager().Rollback(ctx, tx); rollbackErr == nil {
        return nil
    }
    retry++
}
```
this will be better

Reply:
Commit method should Repeated call too

Reply:
ok, you're right